### PR TITLE
Reimplement monitors with tf-framework

### PR DESCRIPTION
**Pull request by tosuke**
Output from acceptance testing:

```
$ MACKEREL_EXPERIMENTAL_TFFRAMEWORK=1 make testacc TESTS='"Acc(DataSource)?MackerelMonitor"'
TF_ACC=1 go test -v ./mackerel/... -run "Acc(DataSource)?MackerelMonitor" -timeout 120m
2024/08/19 11:47:38 [INFO] mackerel: use terraform-plugin-framework based implementation
=== RUN   TestAccDataSourceMackerelMonitorHostMetric
=== PAUSE TestAccDataSourceMackerelMonitorHostMetric
=== RUN   TestAccDataSourceMackerelMonitorConnectivity
=== PAUSE TestAccDataSourceMackerelMonitorConnectivity
=== RUN   TestAccDataSourceMackerelMonitorServiceMetric
=== PAUSE TestAccDataSourceMackerelMonitorServiceMetric
=== RUN   TestAccDataSourceMackerelMonitorExternal
=== PAUSE TestAccDataSourceMackerelMonitorExternal
=== RUN   TestAccDataSourceMackerelMonitorExpression
=== PAUSE TestAccDataSourceMackerelMonitorExpression
=== RUN   TestAccDataSourceMackerelMonitorAnomalyDetection
=== PAUSE TestAccDataSourceMackerelMonitorAnomalyDetection
=== RUN   TestAccDataSourceMackerelMonitorQuery
=== PAUSE TestAccDataSourceMackerelMonitorQuery
=== RUN   TestAccMackerelMonitor_HostMetric
=== PAUSE TestAccMackerelMonitor_HostMetric
=== RUN   TestAccMackerelMonitor_Connectivity
=== PAUSE TestAccMackerelMonitor_Connectivity
=== RUN   TestAccMackerelMonitor_ServiceMetric
=== PAUSE TestAccMackerelMonitor_ServiceMetric
=== RUN   TestAccMackerelMonitor_External
=== PAUSE TestAccMackerelMonitor_External
=== RUN   TestAccMackerelMonitor_Expression
=== PAUSE TestAccMackerelMonitor_Expression
=== RUN   TestAccMackerelMonitor_AnomalyDetection
=== PAUSE TestAccMackerelMonitor_AnomalyDetection
=== RUN   TestAccMackerelMonitor_Query
=== PAUSE TestAccMackerelMonitor_Query
=== CONT  TestAccDataSourceMackerelMonitorHostMetric
=== CONT  TestAccMackerelMonitor_HostMetric
=== CONT  TestAccDataSourceMackerelMonitorExpression
=== CONT  TestAccDataSourceMackerelMonitorServiceMetric
=== CONT  TestAccDataSourceMackerelMonitorExternal
=== CONT  TestAccMackerelMonitor_Expression
=== CONT  TestAccMackerelMonitor_ServiceMetric
=== CONT  TestAccMackerelMonitor_External
--- PASS: TestAccDataSourceMackerelMonitorExpression (9.28s)
=== CONT  TestAccMackerelMonitor_Connectivity
--- PASS: TestAccDataSourceMackerelMonitorExternal (9.47s)
=== CONT  TestAccMackerelMonitor_Query
--- PASS: TestAccDataSourceMackerelMonitorServiceMetric (9.47s)
=== CONT  TestAccMackerelMonitor_AnomalyDetection
--- PASS: TestAccDataSourceMackerelMonitorHostMetric (9.94s)
=== CONT  TestAccDataSourceMackerelMonitorQuery
--- PASS: TestAccMackerelMonitor_Expression (16.00s)
=== CONT  TestAccDataSourceMackerelMonitorAnomalyDetection
--- PASS: TestAccMackerelMonitor_ServiceMetric (16.88s)
=== CONT  TestAccDataSourceMackerelMonitorConnectivity
--- PASS: TestAccMackerelMonitor_HostMetric (17.25s)
--- PASS: TestAccDataSourceMackerelMonitorQuery (7.45s)
--- PASS: TestAccMackerelMonitor_External (17.46s)
--- PASS: TestAccDataSourceMackerelMonitorAnomalyDetection (6.14s)
--- PASS: TestAccMackerelMonitor_Connectivity (13.03s)
--- PASS: TestAccMackerelMonitor_Query (12.97s)
--- PASS: TestAccMackerelMonitor_AnomalyDetection (13.14s)
--- PASS: TestAccDataSourceMackerelMonitorConnectivity (5.99s)
PASS
ok      github.com/mackerelio-labs/terraform-provider-mackerel/mackerel 24.742s
```

### CODE REVIEW

File: internal/provider/resource_mackerel_monitor.go [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+				},
+				"notification_interval": schema.Int64Attribute{
+					Description: schemaMonitorNotificationIntervalDesc,
+					Optional:    true,
+					Computed:    true,
+					Default:     int64default.StaticInt64(0), // TODO(schema upgrade): handle null
+					Validators: []validator.Int64{
+						int64validator.AtLeast(0),
+					},
+				},
```
Comment (tosuke) [inline internal/provider/resource_mackerel_monitor.go:184]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/500723894040269e6bde32c814377b28c15b6f6a/mackerel/resource_mackerel_monitor.go?plain=1#L36-L53

File: internal/provider/resource_mackerel_monitor.go [hunk 2]
```diff
@@ -0,0 +1,647 @@
...
+				"max_check_attempts": schemaMonitorMaxCheckAttemptsAttr(1),
+				"scopes":             schemaMonitorResourceScopesAttr(),
+				"exclude_scopes":     schemaMonitorResourceExcludeScopesAttr(),
+			},
+			Validators: []validator.Object{
+				schemaMonitorResourceThresholdValidator(),
+			},
+		},
+	}
+}
```
Comment (tosuke) [inline internal/provider/resource_mackerel_monitor.go:323]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/500723894040269e6bde32c814377b28c15b6f6a/mackerel/resource_mackerel_monitor.go?plain=1#L54-L106

File: internal/provider/resource_mackerel_monitor.go [hunk 3]
```diff
@@ -0,0 +1,647 @@
...
+					Default:     int64default.StaticInt64(0), // TODO(schema upgrade): handle null
+					Validators:  []validator.Int64{missingDurationValidator},
+				},
+			},
+			Validators: []validator.Object{
+				schemaMonitorResourceThresholdValidator(),
+			},
+		},
+	}
+}
```
Comment (tosuke) [inline internal/provider/resource_mackerel_monitor.go:382]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/500723894040269e6bde32c814377b28c15b6f6a/mackerel/resource_mackerel_monitor.go?plain=1#L107-L132

File: internal/provider/resource_mackerel_monitor.go [hunk 4]
```diff
@@ -0,0 +1,647 @@
...
+						stringvalidator.OneOf("CRITICAL", "WARNING"),
+					},
+				},
+				"scopes":         schemaMonitorResourceScopesAttr(),
+				"exclude_scopes": schemaMonitorResourceExcludeScopesAttr(),
+			},
+		},
+	}
+}
+
```
Comment (tosuke) [inline internal/provider/resource_mackerel_monitor.go:471]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/500723894040269e6bde32c814377b28c15b6f6a/mackerel/resource_mackerel_monitor.go?plain=1#L107-L132

File: internal/provider/resource_mackerel_monitor.go [hunk 5]
```diff
@@ -0,0 +1,647 @@
...
+				"certification_expiration_warning": schema.Int64Attribute{
+					Description: schemaMonitorExternal_CertificationExpirationWarningDesc,
+					Optional:    true,
+					Computed:    true,
+					Default:     int64default.StaticInt64(0),
+				},
+			},
+		},
+	}
+}
```
Comment (tosuke) [inline internal/provider/resource_mackerel_monitor.go:599]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/500723894040269e6bde32c814377b28c15b6f6a/mackerel/resource_mackerel_monitor.go?plain=1#L195-L271

File: internal/provider/resource_mackerel_monitor.go [hunk 6]
```diff
@@ -0,0 +1,647 @@
...
+				"operator": schemaMonitorResourceOperatorAttr(),
+				"warning":  schemaMonitorResourceWarningAttr(),
+				"critical": schemaMonitorResourceCriticalAttr(),
+			},
+			Validators: []validator.Object{
+				schemaMonitorResourceThresholdValidator(),
+			},
+		},
+	}
+}
```
Comment (tosuke) [inline internal/provider/resource_mackerel_monitor.go:408]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/500723894040269e6bde32c814377b28c15b6f6a/mackerel/resource_mackerel_monitor.go?plain=1#L272-L302

File: internal/provider/resource_mackerel_monitor.go [hunk 7]
```diff
@@ -0,0 +1,647 @@
...
+					},
+				},
+				"operator": schemaMonitorResourceOperatorAttr(),
+				"warning":  schemaMonitorResourceWarningAttr(),
+				"critical": schemaMonitorResourceCriticalAttr(),
+			},
+			Validators: []validator.Object{schemaMonitorResourceThresholdValidator()},
+		},
+	}
+}
```
Comment (tosuke) [inline internal/provider/resource_mackerel_monitor.go:443]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/500723894040269e6bde32c814377b28c15b6f6a/mackerel/resource_mackerel_monitor.go?plain=1#L340-L375

File: internal/provider/data_source_mackerel_monitor.go [hunk 8]
```diff
@@ -0,0 +1,201 @@
...
+				Computed:    true,
+			},
+			"is_mute": schema.BoolAttribute{
+				Description: schemaMonitorIsMuteDesc,
+				Computed:    true,
+			},
+			"notification_interval": schema.Int64Attribute{
+				Description: schemaMonitorNotificationIntervalDesc,
+				Computed:    true,
+			},
```
Comment (tosuke) [inline internal/provider/data_source_mackerel_monitor.go:53]: Original schema: https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/500723894040269e6bde32c814377b28c15b6f6a/mackerel/data_source_mackerel_monitor.go?plain=1#L16-L35

File: internal/provider/data_source_mackerel_monitor.go [hunk 9]
```diff
@@ -0,0 +1,201 @@
...
+						"training_period_from": types.Int64Type,
+						"scopes": types.SetType{
+							ElemType: types.StringType,
+						},
+					},
+				},
+			},
+		},
+	}
+}
```
Comment (tosuke) [inline internal/provider/data_source_mackerel_monitor.go:174]: Original schema (almost same as resources): https://github.com/mackerelio-labs/terraform-provider-mackerel/blob/500723894040269e6bde32c814377b28c15b6f6a/mackerel/data_source_mackerel_monitor.go?plain=1#L15-L292

**Review by azukiazusa1 — APPROVED** [review]
lgtm!

**Pull request merged**